// Copyright 2017 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Summary of results for each top-level test.

syntax = "proto3";

package w3c.wptdashboard;
import "bq_table_name.proto";

message TestSummary {
  option (gen_bq_schema.table_name) = "test_summaries";

  // The time associated with "long_wpt_hash".
  int32 wpt_commit_time = 1;
  // A w3c/web-platform-tests Git hash.
  string long_wpt_hash = 2;
  // Top-level test name, as reported in wpt test report file.
  string name = 3;
  // Total number of tests: 1 + [number of sub-tests].
  int32 num_tests_total = 4;
  // Total number of tests passing tests.
  int32 num_tests_passed = 5;
}
